apps/src/lib/ui/accounts/DeleteAccount.jsx

```diff
@@ -107,7 +183,26 @@ export default class DeleteAccount extends React.Component {
           {i18n.deleteAccount()}
         </h2>
         <div style={styles.warning}>
-          {i18n.deleteAccount_warning()}
+          {!this.props.isTeacher && i18n.deleteAccount_studentWarning()}
+          {this.props.isTeacher &&
+            <div>
+              <p>
+                {i18n.deleteAccount_teacherWarning1()}
+                <strong>{i18n.deleteAccount_teacherWarning2()}</strong>
+                {i18n.deleteAccount_teacherWarning3()}
+              </p>
+              <p>
+                {i18n.deleteAccount_teacherWarning4()}
+                <a
+                  href={ADD_A_PERSONAL_LOGIN_HELP_URL}
+                  target="_blank"
+                >
+                  {i18n.deleteAccount_teacherWarning5()}
+                </a>
+                {i18n.deleteAccount_teacherWarning6()}
+              </p>
+            </div>
+          }
         </div>
```

Comment:
This might be a good candidate for extraction to helper components in this file; It's be nice to write
```jsx
{this.props.isTeacher ? <TeacherWarning/> : <StudentWarning/>}
```
and then hide the i18n/formatting mess away.

Reply:
good idea! i'll make that refactor